### PR TITLE
Add random spooky gif feature.

### PR DESCRIPTION
**Pull request by markylon**
Addresses issue #8, adding random spooky gif feature that takes use of the GIPHY API.

To take use of this cog, create a env. variable named GIPHY_TOKEN and use the token you get from https://developers.giphy.com, they have a very simple process of creating an app, just create an account and you should be able to make your app and get the API key.

### CODE REVIEW

File: bot/cogs/gif.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from os import environ
+
+import aiohttp
+from discord.ext import commands
+
+
+class Gif:
```
Comment (heavysaturn) [inline bot/cogs/gif.py]: Gif is perhaps a little too generic. I think SpookyGif is better.

File: bot/cogs/gif.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
```
Comment (heavysaturn) [inline bot/cogs/gif.py]: no blank lines before docstrings.

File: bot/cogs/gif.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    """
+    A cog to fetch a random spooky gif from the web!
+    """
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.GIPHY_TOKEN = environ.get('GIPHY_TOKEN')
+
+    @commands.command()
+    async def gif(self, ctx):
```
Comment (heavysaturn) [inline bot/cogs/gif.py]: this should also have a docstring.

**Comment by markylon** [comment]
Alright, resolved the requested changes. Feel free to test.

File: bot/cogs/gif.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
```
Comment (sco1) [inline bot/cogs/gif.py]: `.spookygif` would probably be a more appropriate command, `.gif` will be fine as an alias

File: bot/cogs/gif.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        Fetches a random gif from the GIPHY API and responds with it.
+        """
+
+        async with aiohttp.ClientSession() as session:
+            params = {'api_key': self.GIPHY_TOKEN, 'tag': 'halloween', 'rating': 'g'}
+            # Make a GET request to the Giphy API to get a random halloween gif.
+            async with session.get('http://api.giphy.com/v1/gifs/random', params=params) as resp:
+                data = await resp.json()
+            url = data['data']['url']
+            await ctx.send(url)
```
Comment (sco1) [inline bot/cogs/gif.py]: Let's have a more festive response than just a URL dump

**Comment by sco1** [comment]
Any progress on the requested changes?

**Comment by markylon** [comment]
@sco1 Sorry, I'm currently on vacation and have no PC access. Will get this done soon.

File: bot/cogs/gif.py (hunk 4 above)
Comment (heavysaturn) [reply bot/cogs/gif.py]: .gif as an alias sorta conflicts with our new idea of how this bot should work. doing `.sgif` instead.

**Pull request merged**